modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala

```diff
@@ -46,19 +47,21 @@ object Main extends TessellationIOApp[Run]("", "DAG L1 node", version = BuildInf
           .start(storages, services, queues, p2pClient, rumorHandler, nodeId, cfg)
           .asResource
 
-        api = HttpApi.make[IO](storages, queues, keyPair.getPrivate, services, sdkPrograms, sdk.nodeId)
+        api = HttpApi.make[IO](storages, queues, keyPair.getPrivate, services, programs, sdk.nodeId)
         _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.http.publicHttp, api.publicApp)
         _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.http.p2pHttp, api.p2pApp)
         _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.http.cliHttp, api.cliApp)
         stateChannel = new StateChannel(cfg, keyPair, p2pClient, queues, nodeId, services, storages, validators)
         _ <- {
           method match {
-            case _: RunInitialValidator =>
-              storages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin) >>
+            case cfg: RunInitialValidator =>
```

Comment:
That peer should has Ready state, not ReadyForJoining

Reply:
and I'd wrap peer discovery into own statuses (see genesis on L0) but it's up to you

Reply:
Not sure if it needs to be Ready. It's ready to join (or be joined) then it creates the session and changes to Ready just like other nodes do at the first peer joining. Anyway that's not the scope of this PR so if we want to we can change it in a separate one. Regarding own statuses even if I created own list of them the current ones seem to be baked in into the cluster functionality in SDK pretty tightly. We could maybe have some more generic statuses added to SDK so that state channels could use them. I can wrap the L0 peer discovery in switch from `Initial` to `ReadyToJoin` but I guess that's not what you meant.

Reply:
I believe it is in the scope of this PR because you're adding peer discovery. If node starts as an Initial peer, it needs to start a session and change its status to Ready, and accept incoming joins. 
Nodes that are starting as a validator needs to set status as ReadyToJoin, so then they can be connected to the network